src/Entity/EntityId.php

```diff
@@ -30,6 +37,68 @@ public function getSerialization() {
 	}
 
 	/**
+	 * Returns an array with 3 elements: the foreign repository name as the first element, the local ID as the last
+	 * element and everything that is in between as the second element.
+	 *
+	 * @param $serialization
+	 * @return array
+	 */
+	public static function splitSerialization( $serialization ) {
+		$parts = explode( ':', $serialization );
+		$localPart = array_pop( $parts );
+		$repoName = array_shift( $parts ) ?: '';
+		$prefixRemainder = implode( ':', $parts );
+
+		return array( $repoName, $prefixRemainder, $localPart );
+	}
```

Comment:
Positional arrays act as tuples, which seem to be a standard feature in several languages, especially functional ones. They are ideomatic in php via the list() syntax.

Also, returning a list seems fine here, since this is a method that splits something.

I agree with jakobw: We could add keys and make the result associative - if we could think of good names for the parts. "start", "middle", "end" are not very helpful...